ui/src/shared/components/Dygraph.js

```diff
@@ -149,6 +153,40 @@ export default React.createClass({
     delete this.dygraph
   },
 
+  // Only update the component if props other than timeRange have updateDashboard.
+  // If only timeRange has changed, only reset the zoom.
+  // TODO: possibly try to use dateWindow and a zoomCallback to updateOptions
+  // in comonentDidUpdate so that zoom and redraw happen at the same time
+  shouldComponentUpdate(nextProps) {
+    const timeRangeChanged =
+      nextProps.timeRange.lower !== this.props.timeRange.lower ||
+      nextProps.timeRange.upper !== this.props.timeRange.upper
```

Comment:
Can this check be rolled into L172? Maybe:

1. Compile list of properties that have changed
2. If list includes `lower` or `upper`, then resetZoom
3. Remove `lower` and `upper`
4. Return length of list